File: 6/first.js
#!/usr/bin/env node

'use strict'

const fs = require('fs');
const _ = require('lodash');

const s=  fs.readFileSync(process.argv[2] || "input.txt").toString();
const groups = s.split('\n\n');

let ans = 0
_.map(groups, group => {
    let lines = _.filter(group.split('\n'))
    let set = {} 
    console.log(lines);
    _.map(lines, line => {
        _.map(line, ch => {
            set[ch] = ~~set[ch] + 1;
            //console.log(ch, set[ch], lines);
            if (set[ch] == lines.length) {
                ans++;
                console.log(ch, set[ch], set, lines);
            }
        })
    })
    // console.log(Object.keys(set).length);
    // ans += Object.keys(set).length;
})

console.log(ans);

File: 3/first.py
#!/usr/bin/env python3

data = open('input.txt').read().split('\n')

data = [ x for x in data if len(x) > 0 ]

WIDTH=len(data[0])

def filt(ts):
  return len([x for x in ts if x == '#'])

def main():
  trees = [ data[i][3 * i % WIDTH] for i in range(len(data)) ]
  print(filt(trees))

  vals = [
    filt([ data[i][1 * i % WIDTH] for i in range(len(data)) ]),
    filt([ data[i][3 * i % WIDTH] for i in range(len(data)) ]),
    filt([ data[i][5 * i % WIDTH] for i in range(len(data)) ]),
    filt([ data[i][7 * i % WIDTH] for i in range(len(data)) ]),
    filt([ data[2 * i][1 * i % WIDTH] for i in range(len(data)//2) ]),
  ]
  print(vals)
  print(vals[0] * vals[1] * vals[2] * vals[3] * vals[4])




main()

